vm_core.h

```diff
@@ -108,7 +108,78 @@ extern int ruby_assert_critical_section_entered;
 
 #if USE_SHARED_GC
 typedef struct gc_function_map {
```

Comment:
Don't hate me but, do you need function pointers for this? Can't you include different files based on preprocessor definitions (include the GC at compilation time?). Is this being done so you can have the GC in an external dynamic library and select it with a command-line flag?

Reply:
Yes, the goal of the project is to load different GCs at runtime. It is difficult to compile many different Rubies to use in production, but much easier to load different GC implementations using an environment variable.